tools/systematic.py

```diff
@@ -116,107 +125,221 @@ def get_systematic_measured_values_only(options, syst_x_secs_and_errs):
+    normalised_systematic_uncertainty_x_sections['central'] = central_measurement
+    unfolded_normalised_systematic_uncertainty_x_sections['central'] = central_measurement
+
+    for systematic, variation in list_of_systematics.iteritems():
+        if (systematic == 'PDF'):
+            syst_unc_x_sec, unf_syst_unc_x_sec = read_normalised_xsection_systematics(options, variation, is_multiple_sources=True)
+            normalised_systematic_uncertainty_x_sections[systematic] = []
+            unfolded_normalised_systematic_uncertainty_x_sections[systematic] = []
+            for weight, vals in syst_unc_x_sec.iteritems():
+                normalised_systematic_uncertainty_x_sections[systematic].append([weight, vals])
+            for weight, vals in unf_syst_unc_x_sec.iteritems():
+                unfolded_normalised_systematic_uncertainty_x_sections[systematic].append([weight, vals])
+
+        elif (systematic == 'TTJets_envelope'):
+            syst_unc_x_sec, unf_syst_unc_x_sec = read_normalised_xsection_systematics(options, variation, is_multiple_sources=True)
+            env_lower, env_upper = get_scale_envelope(options, syst_unc_x_sec)
+            unf_env_lower, unf_env_upper = get_scale_envelope(options, unf_syst_unc_x_sec)
+            normalised_systematic_uncertainty_x_sections[systematic] = [
+                env_lower,
+                env_upper,
+            ]
+            unfolded_normalised_systematic_uncertainty_x_sections[systematic] = [
+                unf_env_lower,
+                unf_env_upper,
+            ]
+        # elif (systematic == 'TTJets_alphaS'): continue
+        else :
+            syst_unc_x_sec, unf_syst_unc_x_sec = read_normalised_xsection_systematics(options, variation)
+            normalised_systematic_uncertainty_x_sections[systematic] = [
+                syst_unc_x_sec['upper'], 
+                syst_unc_x_sec['lower'],
+            ]
+            unfolded_normalised_systematic_uncertainty_x_sections[systematic] = [
+                unf_syst_unc_x_sec['upper'], 
+                unf_syst_unc_x_sec['lower'],
+            ]
 
     return normalised_systematic_uncertainty_x_sections, unfolded_normalised_systematic_uncertainty_x_sections
 
+
+def get_scale_envelope(options, d_syst):
+    '''
+    Calculate the scale envelope for the renormalisation/factorisation/combined systematic uncertainties
+    For all up variations in a bin keep the highest
+    For all down variations in a bin keep the lowest
+    '''
+    down_variations = []
+    up_variations = []
+    envelope_up = []
+    envelope_down = []
+
+    # Separate into up/down scale variations
+    for scale_variation in d_syst:
+        if 'down' in scale_variation:
+            down_variations.append(d_syst[scale_variation])
+        elif 'up' in scale_variation:
+            up_variations.append(d_syst[scale_variation])
+
+    # find min/max
+    for v1, v2, v3 in zip (up_variations[0], up_variations[1], up_variations[2]):
```

Comment:
So the d_syst is a dictionary containing all Q2 scale variations (muR up/down, muF up/down muRmuF up/down) of which all the up(down) variations are stored in up(down)_variations[]. As these are the only variations passed to the function the up(down)_variations will always be length 3. I will update the doc to reflect this